### PR TITLE
Append Doctrine proxy class names to list of forbidden class names to…

### PR DESCRIPTION
**Pull request by kamil-zacek**
… disallow direct use of Doctrine proxy classes.

### CODE REVIEW

File: tests/Classes/DoctrineProxyForbiddenClassNamesExtensionTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * @extends RuleTestCase<InstantiationRule>
+ */
+class DoctrineProxyForbiddenClassNamesExtensionTest extends RuleTestCase
+{
+	protected function getRule(): Rule
+	{
+		$reflectionProvider = $this->createReflectionProvider();
+
+		return new InstantiationRule(
```
Comment (ondrejmirtes) [inline tests/Classes/DoctrineProxyForbiddenClassNamesExtensionTest.php]: I love this! On this line it makes more sense to do `self::getContainer()->getByType(InstantiationRule::class)`. Thanks!

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Try putting `phpstan/phpstan:1.10.x-dev` in `composer.json` temporarily to verify the build will pass.

File: tests/Classes/DoctrineProxyForbiddenClassNamesExtensionTest.php (hunk 1 above)
Comment (kamil-zacek) [reply tests/Classes/DoctrineProxyForbiddenClassNamesExtensionTest.php]: OK, but there is an phpstan error - I don't know how / whether to solve it:

Error: Accessing PHPStan\Rules\Classes\InstantiationRule::class is not covered by backward compatibility promise. The class might change in a minor PHPStan version.
 ------ ----------------------------------------------------------------------------- 
  Line   tests/Classes/DoctrineProxyForbiddenClassNamesExtensionTest.php              
 ------ ----------------------------------------------------------------------------- 
  [18](https://github.com/phpstan/phpstan-doctrine/actions/runs/8359861123/job/22884119180#step:7:19)     Accessing PHPStan\Rules\Classes\InstantiationRule::class is not              
         covered by backward compatibility promise. The class might change in         
         a minor PHPStan version.                                                     
         💡 If you think it should be covered by backward compatibility                
            promise, open a discussion:                                               
         💡 https://github.com/phpstan/phpstan/discussions                             
         💡                                                                            
         💡 See also:                                                                  
         💡                                                                            
            https://phpstan.org/developing-extensions/backward-compatibility-promise

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Awesome, thank you!

**Pull request merged**